client/lib/signup/step-actions.js

```diff
@@ -9,20 +9,102 @@ import async from 'async';
 /**
  * Internal dependencies
  */
+import { cartItems } from 'lib/cart-values';
 import wpcom from 'lib/wp' ;
 const sites = require( 'lib/sites-list' )();
 const user = require( 'lib/user' )();
 import { getSavedVariations } from 'lib/abtest';
 import SignupCart from 'lib/signup/cart';
+import { startFreeTrial } from 'lib/upgrades/actions';
+
+function addDomainItemsToCart( callback, dependencies, { domainItem, googleAppsCartItem, isPurchasingItem, siteUrl, themeSlug, themeItem } ) {
+	wpcom.undocumented().sitesNew( {
+		blog_name: siteUrl,
+		blog_title: siteUrl,
+		options: {
+			theme: dependencies.theme
+		},
+		validate: false,
+		find_available_url: isPurchasingItem
+	}, function( error, response ) {
+		if ( error ) {
+			callback( error );
+
```

Comment:
I removed that in https://github.com/Automattic/wp-calypso/pull/4147/commits/7597d0576d8bed67643119938a9eda7e2b3f0ab3 since it was highly ambiguous. Because of that and missing docblock comments I had to spent a significant amount of time trying to figure out whether these actions and steps actually use what was returned. 

The empty line is to improve readability by highlighting the return statement: that's an important piece here in my opinion since it's an exit point.